### PR TITLE
Add point_xyz as pre-defined geometry model

### LINKED ISSUE #655: Proposal: Can we have point_xyz?
`point_xy` made everyone on the plane happy. Can we make everyone in 3D space happy with `point_xyz`?

### PR DESCRIPTION
**Pull request by digu-007**
This PR fixes #655. I have added the `model::d3::point_xyz` with example.

I have verified the documentation, here is the result:
point_xyz_1
point_xyz_2

### CODE REVIEW

**Comment by digu-007** [comment]
@mloskot Can you review the PR?

File: doc/reference/geometries/point_xyz.qbk [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  Copyright (c) 2020 Digvijay Janartha, Hamirpur, India.
+
+  Use, modification and distribution is subject to the Boost Software License,
+  Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+  http://www.boost.org/LICENSE_1_0.txt)
+=============================================================================/]
+
+[heading Examples]
+[point_xyz]
+[point_xyz_output]
```
Comment (mloskot) [inline doc/reference/geometries/point_xyz.qbk]: Newline is missing

**Comment by digu-007** [comment]
@mloskot I have added simple unit test for `point_xyz`.
> I realised there is no such test for `point_xy`

I couldn't find them either. If the changes gets accepted here, then I can add such tests for `point_xy` also.

**Comment by awulkiew** [comment]
@mloskot I don't recall moving the tests. I think there were no explicit tests for the model. However they are used in all other tests, including `point_xy`.

**Comment by mloskot** [comment]
@awulkiew Right. I just had an impression there used to be basic unit tests for models, before they actually got covered by tests of algorithms, etc.

I, personally, see value in basic unit tests. They clearly show what operations on model classes are actually tested. Not something that can be easily displayed from all the complex tests, unless one knows what to grep through the code for. If there is no objection, e.g. it would hit CI performance, I think adding tests for models is a good thing, no?

**Comment by awulkiew** [comment]
@mloskot > I think adding tests for models is a good thing, no?

Of course, I was just explaining the current state.

**Comment by mloskot** [comment]
@awulkiew Sure, thanks! I have opened related issue in #673

**Comment by mloskot** [comment]
@digu-007 Thanks for your contribution.

**Comment by digu-007** [comment]
@mloskot @awulkiew @vissarion Thanks for merging and guidance :)

**Pull request merged**